OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -1524,9 +1523,36 @@ public static DreamValue NativeProc_regex(DreamObject instance, DreamObject usr,
 
             DreamList list = DreamList.Create();
 
-            list.AddValue(new DreamValue(c.RByte));
-            list.AddValue(new DreamValue(c.GByte));
-            list.AddValue(new DreamValue(c.BByte));
+            switch(space) {
+                case 0: //rgb
+                    list.AddValue(new DreamValue(c.RByte));
+                    list.AddValue(new DreamValue(c.GByte));
+                    list.AddValue(new DreamValue(c.BByte));
+                    break;
+                case 1: //hsv
+                    Vector4 hsvcolor = Color.ToHsv(c);
+                    list.AddValue(new DreamValue(hsvcolor.X * 360));
+                    list.AddValue(new DreamValue(hsvcolor.Y * 100));
+                    list.AddValue(new DreamValue(hsvcolor.Z * 100));
+                    break;
+                case 2: //hsl
+                    Vector4 hslcolor = Color.ToHsl(c);
+                    list.AddValue(new DreamValue(hslcolor.X * 360));
+                    list.AddValue(new DreamValue(hslcolor.Y * 100));
+                    list.AddValue(new DreamValue(hslcolor.Z * 100));
+                    break;
+                case 3: //hcy
+                    /// TODO Figure out why the chroma for #ca60db is 48 instead of 68
+                    throw new NotImplementedException("HCY Colorspace is not impelemented");
+                    /*
+                    Vector4 hcycolor = Color.ToHcy(c);
+                    list.AddValue(new DreamValue(hcycolor.X * 360));
+                    list.AddValue(new DreamValue(hcycolor.Y * 100));
+                    list.AddValue(new DreamValue(hcycolor.Z * 100));
+                    */
+                default:
+                    throw new NotImplementedException("Colorspace is not impelemented");
```

Comment:
Isn't HCY the only unimplemented, which has its own NotImplementedException? This should be a bad arg error (and also should tell you what the bad arg's value is)